src/databricks/labs/ucx/framework/workflows.py

```diff
@@ -0,0 +1,27 @@
+from collections.abc import Iterable
+
+from databricks.labs.ucx.contexts.workflow_task import RuntimeContext
+from databricks.labs.ucx.framework.tasks import Task, job_task
+
+
+class Workflow:
+    def __init__(self, name: str):
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    def tasks(self) -> Iterable[Task]:
+        # return __task__ from every method in this class that has this attribute
+        for attr in dir(self):
+            if attr.startswith("_"):
+                continue
+            fn = getattr(self, attr)
+            if hasattr(fn, "__task__"):
+                yield fn.__task__
+
+    @job_task
+    def parse_logs(self, ctx: RuntimeContext):
```

Comment:
@nfx : Need to add the `job_task` here as well besides adding the parse logs in the `DeployedWorkflows` to access it at runtime

Reply:
@JCZuurmond what i meant - you should add an if-statement to `databricks.labs.ucx.runtime.Workflows.trigger`. Producing logs for this task is pointless, so it doesn't need to go through the whole `TaskLogger` context manager. And then you don't need to move things around.

<img width="633" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/7e4c0d23-f96a-405d-a26a-634a3f1b650b">